### ISSUE TITLE
PushNotification.init() call fails when running app with latest plugin version

### ISSUE THREAD

**Issue by VirtuallyInvincible** [phonegap/phonegap-plugin-push#1866]
### Expected Behaviour
The init() function call does not fail.

### Actual Behaviour
PushNotification.init() fails and execution of the program halts

### Reproduce Scenario (including but not limited to)
Hello.

I opened a blank cordova application. I set the cordova version to 6.4.0. The cordova-android version is 6.2.3. I put the following code in the index.js file of the project:
```js
function onDeviceReady() {
        // Handle the Cordova pause and resume events
        document.addEventListener( 'pause', onPause.bind( this ), false );
        document.addEventListener( 'resume', onResume.bind( this ), false );
        
        // TODO: Cordova has been loaded. Perform any initialization that requires Cordova here.
        var parentElement = document.getElementById('deviceready');
        var listeningElement = parentElement.querySelector('.listening');
        var receivedElement = parentElement.querySelector('.received');
        listeningElement.setAttribute('style', 'display:none;');
        receivedElement.setAttribute('style', 'display:block;');

        console.log("Starting init()...");
        var push = PushNotification.init({
          android: {
            vibrate: true,
            messageKey: "This is a message",
            titleKey: "This is a title"
          },
          browser: {},
          ios: {},
          windows: {}
        });
        console.log("init() completed!");

        console.log("Starting hasPermission()...");
        PushNotification.hasPermission(function (data) {
          console.log("data.isEnabled = " + data.isEnabled);
        });
        console.log("hasPermission() completed!");

        console.log("Starting push.on('registration')...");
        push.on('registration', function (data) {
          console.log("push.on('registration') called");
          console.log(data.registrationId);
          console.log(data.registrationType);
        });
        console.log("push.on('registration') completed!");

        console.log("Starting push.on('notification')...");
        push.on('notification', function (data) {
          console.log("push.on('notification') called");
          console.log("data.message = " + data.message);
          console.log("data.title = " + data.title);
          console.log("data.count = " + data.count);
          console.log("data.sound = " + data.sound);
          console.log("data.image = " + data.image);
          console.log("data.additionalData = " + data.additionalData);
        });
        console.log("push.on('notification') completed!");

        console.log("Starting push.on('error')...");
        push.on('error', function (e) {
          console.log("push.on('error') called");
          console.log(e.message);
        });
        console.log("push.on('error') completed!");

        console.log("Starting subscribe()...");
        push.subscribe('example', function () {
          console.log('Subscription successful!');
        }, function (e) {
          console.log('Subscription error:');
          console.log(e);
        });
        console.log("subscribe() completed!");
    };
```

When using version 2.0.0-rc5 of the plugin, the following is printed to the javascript console window of Visual Studio:
Starting init()...
index.js (22,17)
Current window: 

As you can see, execution halts as soon as init() is called. If I use version 1.10.5 of the plugin, the logs are as below:
Starting init()...
index.js (22,17)
init() completed!
index.js (33,17)
Starting hasPermission()...
index.js (35,17)
hasPermission() completed!
index.js (39,17)
Starting push.on('registration')...
index.js (41,17)
push.on('registration') completed!
index.js (47,17)
Starting push.on('notification')...
index.js (49,17)
push.on('notification') completed!
index.js (59,17)
Starting push.on('error')...
index.js (61,17)
push.on('error') completed!
index.js (66,17)
Starting subscribe()...
index.js (68,17)
subscribe() completed!
index.js (75,17)
data.isEnabled = true
index.js (37,19)
Subscription successful!
index.js (70,19)
push.on('error') called
index.js (63,19)
No value for senderID
index.js (64,19)
Current window:

In older versions, the code fails because I haven't inserted the senderID value, but in the latest version, I don't need to do so. I only need to put the google-services.json file in the project's root directory. I do have the file as requested, yet the init() method fails.

#### Steps to Reproduce 
1. Start new project in Visual Studio 2015.
2. Run npm install -g cordova@6.4.0 in the project's root directory in command line window and change Cordova version of the project to 6.4.0 in config.xml file.
3. Update cordova-android version for the project by running cordova platform update android@6.2.3 in command line and verify that the project was updated with the corresponding cordova-android version by running cordiva platform -ls.
4. Register to Firebase and add google-services.json to root of project (same level as package.json file).
5. Install the latest version of phonegap-plugin-push (version 2.0.0-rc5). Verify that the plugin was added to the plugins directory of the project.
6. Open index.js file and add the code above in the onDeviceReady() method. Of course, register to the 'deviceready' event and point to the onDeviceReady() method.
7. Connect device to computer and run the project.

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
Device: Android 7.0.0
Cordova-android version installed for project: 6.2.3

#### (Android) What device vendor (e.g. Samsung, HTC, Sony...)
Samsung

#### Cordova CLI version and cordova platform version

    cordova --version                                    #6.4.0
    cordova platform version android                     #6.2.3

#### Plugin version

    phonegap-plugin-push   #2.0.0-rc5

#### Sample Code that illustrates the problem
See reproduce section
![cmd window with android version installed](https://user-images.githubusercontent.com/14920433/28408427-1636e4b0-6d40-11e7-95ce-c4f0667dd232.png)
![screenshot with cordova version used for project](https://user-images.githubusercontent.com/14920433/28408426-1635cec2-6d40-11e7-8e5a-3e4f9419ae5b.png)
![screenshot with the plugin version installed](https://user-images.githubusercontent.com/14920433/28408428-16373988-6d40-11e7-8a04-f47e39752fee.png)

#### Logs taken while reproducing problem
See reproduce section

**Comment by fredgalvao** [comment]
It's possibly a minor detail, but could you try reproducing all of that with a newer version of `node`?

**Comment by macdonst** [comment]
@VirtuallyInvincible my educated guess is this is a Visual Studio problem. This plugin uses cocoapods which VS probably has no idea how to handle. Check your projects `platforms/android/assets/www/plugins/` folder to see if the plugin is even being included correctly which I'm guessing it is not.

**Comment by VirtuallyInvincible** [comment]
Thanks for the answers. I use an old version of node because I had an issue with deploying projects and I found out this version of node is the most basic version that supports all versions of Cordova and indeed it solved my problems despite the warnings. I'll check about cocoapods

**Comment by VirtuallyInvincible** [comment]
I'm using a plugin that's supposed to add support for cocoapods:
https://github.com/blakgeek/cordova-plugin-cocoapods-support
It didn't add anything to the android platform and the platforms/android/assets/www/ directory has no plugins folder inside of it. However, cocoapods is for IOS, right? I'm interested to support only Android for now. Must I add handle cocoapods?

Thanks!

**Comment by macdonst** [comment]
@VirtuallyInvincible does your config.xml have an ios section in it? If it does I'd do a `cordova platform rm ios` and try adding the push plugin again.

**Comment by VirtuallyInvincible** [comment]
It does. I ran the command but still I see ios sections in config.xml. Here's my config.xml after running the command:
`<?xml version="1.0" encoding="utf-8"?>
<widget defaultlocale="en-US" id="io.cordova.myapp303cd0" version="1.0.0" xmlns="http://www.w3.org/ns/widgets" xmlns:cdv="http://cordova.apache.org/ns/1.0" xmlns:vs="http://schemas.microsoft.com/appx/2014/htmlapps">
  <name>BlankCordovaApp4</name>
  <description>A blank project that uses Apache Cordova to help you build an app that targets multiple mobile platforms: Android, iOS, Windows, and Windows Phone.</description>
  <author email="dev@cordova.apache.org" href="http://cordova.io">Apache Cordova Team </author>
  <content src="index.html" />
  <access origin="*" />
  <vs:features />
  <preference name="SplashScreen" value="screen" />
  <preference name="windows-target-version" value="8.1" />
  <plugin name="cordova-plugin-whitelist" version="1" />
  <allow-intent href="http://*/*" />
  <allow-intent href="https://*/*" />
  <allow-intent href="tel:*" />
  <allow-intent href="sms:*" />
  <allow-intent href="mailto:*" />
  <allow-intent href="geo:*" />
  <platform name="android">
    <allow-intent href="market:*" />
  </platform>
  <platform name="ios">
    <allow-intent href="itms:*" />
    <allow-intent href="itms-apps:*" />
  </platform>
  <platform name="android">
    <icon density="ldpi" src="res/icons/android/icon-36-ldpi.png" />
    <icon density="mdpi" src="res/icons/android/icon-48-mdpi.png" />
    <icon density="hdpi" src="res/icons/android/icon-72-hdpi.png" />
    <icon density="xhdpi" src="res/icons/android/icon-96-xhdpi.png" />
  </platform>
  <platform name="ios">
    <icon height="180" src="res/icons/ios/icon-60-3x.png" width="180" />
    <icon height="60" src="res/icons/ios/icon-60.png" width="60" />
    <icon height="120" src="res/icons/ios/icon-60-2x.png" width="120" />
    <icon height="76" src="res/icons/ios/icon-76.png" width="76" />
    <icon height="152" src="res/icons/ios/icon-76-2x.png" width="152" />
    <icon height="40" src="res/icons/ios/icon-40.png" width="40" />
    <icon height="80" src="res/icons/ios/icon-40-2x.png" width="80" />
    <icon height="57" src="res/icons/ios/icon-57.png" width="57" />
    <icon height="114" src="res/icons/ios/icon-57-2x.png" width="114" />
    <icon height="72" src="res/icons/ios/icon-72.png" width="72" />
    <icon height="144" src="res/icons/ios/icon-72-2x.png" width="144" />
    <icon height="29" src="res/icons/ios/icon-small.png" width="29" />
    <icon height="58" src="res/icons/ios/icon-small-2x.png" width="58" />
    <icon height="50" src="res/icons/ios/icon-50.png" width="50" />
    <icon height="100" src="res/icons/ios/icon-50-2x.png" width="100" />
  </platform>
  <platform name="windows">
    <icon height="150" src="res/icons/windows/Square150x150Logo.scale-100.png" width="150" />
    <icon height="360" src="res/icons/windows/Square150x150Logo.scale-240.png" width="360" />
    <icon height="30" src="res/icons/windows/Square30x30Logo.scale-100.png" width="30" />
    <icon height="310" src="res/icons/windows/Square310x310Logo.scale-100.png" width="310" />
    <icon height="44" src="res/icons/windows/Square44x44Logo.scale-100.png" width="44" />
    <icon height="106" src="res/icons/windows/Square44x44Logo.scale-240.png" width="106" />
    <icon height="70" src="res/icons/windows/Square70x70Logo.scale-100.png" width="70" />
    <icon height="71" src="res/icons/windows/Square71x71Logo.scale-100.png" width="71" />
    <icon height="170" src="res/icons/windows/Square71x71Logo.scale-240.png" width="170" />
    <icon height="50" src="res/icons/windows/StoreLogo.scale-100.png" width="50" />
    <icon height="120" src="res/icons/windows/StoreLogo.scale-240.png" width="120" />
    <icon height="150" src="res/icons/windows/Wide310x150Logo.scale-100.png" width="310" />
    <icon height="360" src="res/icons/windows/Wide310x150Logo.scale-240.png" width="744" />
  </platform>
  <platform name="wp8">
    <icon height="99" src="res/icons/wp8/ApplicationIcon.png" width="99" />
    <icon height="159" src="res/icons/wp8/Background.png" width="159" />
  </platform>
  <platform name="android">
    <splash density="land-hdpi" src="res/screens/android/screen-hdpi-landscape.png" />
    <splash density="land-ldpi" src="res/screens/android/screen-ldpi-landscape.png" />
    <splash density="land-mdpi" src="res/screens/android/screen-mdpi-landscape.png" />
    <splash density="land-xhdpi" src="res/screens/android/screen-xhdpi-landscape.png" />
    <splash density="port-hdpi" src="res/screens/android/screen-hdpi-portrait.png" />
    <splash density="port-ldpi" src="res/screens/android/screen-ldpi-portrait.png" />
    <splash density="port-mdpi" src="res/screens/android/screen-mdpi-portrait.png" />
    <splash density="port-xhdpi" src="res/screens/android/screen-xhdpi-portrait.png" />
  </platform>
  <platform name="ios">
    <splash height="480" src="res/screens/ios/screen-iphone-portrait.png" width="320" />
    <splash height="960" src="res/screens/ios/screen-iphone-portrait-2x.png" width="640" />
    <splash height="1024" src="res/screens/ios/screen-ipad-portrait.png" width="768" />
    <splash height="2048" src="res/screens/ios/screen-ipad-portrait-2x.png" width="1536" />
    <splash height="768" src="res/screens/ios/screen-ipad-landscape.png" width="1024" />
    <splash height="1536" src="res/screens/ios/screen-ipad-landscape-2x.png" width="2048" />
    <splash height="1136" src="res/screens/ios/screen-iphone-568h-2x.png" width="640" />
    <splash height="1334" src="res/screens/ios/screen-iphone-portrait-667h.png" width="750" />
    <splash height="2208" src="res/screens/ios/screen-iphone-portrait-736h.png" width="1242" />
    <splash height="1242" src="res/screens/ios/screen-iphone-landscape-736h.png" width="2208" />
  </platform>
  <platform name="windows">
    <splash height="300" src="res/screens/windows/SplashScreen.scale-100.png" width="620" />
    <splash height="1920" src="res/screens/windows/SplashScreen.scale-240.png" width="1152" />
    <splash height="1920" src="res/screens/windows/SplashScreenPhone.scale-240.png" width="1152" />
  </platform>
  <platform name="wp8">
    <splash height="1280" src="res/screens/wp8/SplashScreenImage.png" width="768" />
  </platform>
  <plugin name="phonegap-plugin-push" src="https://github.com/phonegap/phonegap-plugin-push" version="2.0.0-rc5" />
  <plugin name="cordova-plugin-cocoapod-support" spec="~1.3.0" />
  <engine name="android" spec="~6.2.3" />
</widget>`

Should I manually remove any ios section from the file and then try installing again?

**Comment by macdonst** [comment]
@VirtuallyInvincible yeah, try removing and of the ios platform tags in your config.xml and that `cordova-plugin-cocoapod-support` plugin is not required either.

**Comment by jcesarmobile** [comment]
You have to put a resource-file tag in your config.xml to copy the google-services.json to the project, putting it on the root is not enough.

Also, that requires Cordova CLI 7.x.x, @macdonst we should change the engines tag to require that version, right now it allows >= 6.4.0, but that version doesn't have the resource-file tag

**Comment by HugoHeneault** [comment]
I'm having the same problem as you @VirtuallyInvincible 
Did you find something?

**Comment by VirtuallyInvincible** [comment]
Not yet.

**Comment by VirtuallyInvincible** [comment]
Nope. Can't find a solution, even after reinstalling Visual Studio. Version 2 simply doesn't work for me.

**Comment by macdonst** [comment]
@VirtuallyInvincible I think you'd be better off trying VS Code instead of VS. AFAIK Microsoft is rolling back Cordova features in VS in favour of Xamarin.

**Comment by macdonst** [comment]
Closing, inactivity. Please comment/re-open if you have more details.

**Comment by ArvidSmit** [comment]
@VirtuallyInvincible I came across the same problem. 

In my case I upgraded from cordova 6.2.3 to cordova 6.3.0

When building for IOS i found out that in the new version cordova-plugin-console wasn't necessary anymore so i removed the plugin.

Then I found out that in 6.3.0 out also needed to remove cordova-plugin-compat which conflicted with depencies from some other plugins I use. 

So I turned out the downgrade to cordova 6.2.3 without reinstalling cordova-plugin-console.

I then had the same problems as you. PushNotification.init would not finish and the app stopped right there.

After installing cordova-plugin-console, everything started working again!

Hop this will help solve your problem!
